### ISSUE TITLE
Replace dict.has_key() with the 'in' keyword, for twisted/docs

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8359]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8359|
|Type|enhancement|
|Created|2016-05-26 17:24:49Z|
|Branch|https://github.com/twisted/twisted/tree/nohaskey-8359|

In PEP 0290, has_key() was deprecated in Python 2.2 and higher: 
* https://www.python.org/dev/peps/pep-0290/#testing-dictionary-membership
* https://docs.python.org/2.2/whatsnew/node4.html

In Python 3, dict.has_key() was removed:
* https://docs.python.org/3.0/whatsnew/3.0.html#builtins

PEP 0290 says that for testing dictionary membership, this:

```
if d.has_key(k):
```

should be changed to this:

```
if k in d:
```

<details><summary>Searchable metadata</summary>

```
trac-id__8359 8359
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__nohaskey_8359 nohaskey-8359
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1464283489798785 1464283489798785
changetime__1474198049187875 1474198049187875
version__None None
owner__Wilfredo_S_nchez_Vega__wsanchez_____ Wilfredo Sánchez Vega <wsanchez@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/75

This was pulled out from [ticket:4053]

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_4"></a>|@wsanchez set owner to @wsanchez<br>@wsanchez set status to `closed`|
|-|-|

In changeset c1bc604a2793426e85e2cbd5dc2ccf20d7aca6f8
```
#!CommitTicketReference repository="" revision="c1bc604a2793426e85e2cbd5dc2ccf20d7aca6f8"
Merge pull request #75 from rodrigc/has_key1

Merge nohaskey-8359: Brief description

Author: rodrigc
Reviewer: wsanchez
Fixes: #8359

Replace dict.has_key() with the 'in' keyword, for twisted/docs
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
